### PR TITLE
feat: customer managed keys (CMEK)

### LINKED ISSUE #272: Add CMEK support to Bigtable
Feature checklist:

- [x] Create `EncryptionInfo` class
- [x] Create `Status` class for `EncryptionInfo.encryption_status` field
- [x] Add `encryption_info` field to Backup
- [x] Add `get_encryption_info` method to `Table`
- [x] Add `kms_key_name` field to Cluster

See also the [Java reference implementation](https://github.com/googleapis/java-bigtable/pull/656).

### PR DESCRIPTION
**Pull request by chrisrossi**
Implement customer managed keys (CMEK) feature.

Closes #272

### CODE REVIEW

**Comment by chrisrossi** [comment]
System tests only work with a particular project and a particular key. I'll need to get it creating its own key ring and key to use.

CMEK system tests are also, literally, just a copy/paste of the existing system tests with the `kms_key_name` argument added at cluster creation time. There's probably a better way that doesn't replicate so much code. But maybe that's fine.

I need to flesh out a wrapper class, still, for the `Status` PB that is the value for `EncryptionInfo.encryption_status`. ATM, it just returns the raw PB.

File: google/cloud/bigtable/cluster.py
Comment (kolea2) [inline google/cloud/bigtable/cluster.py:163]: what does this default to if this condition is not met?

File: google/cloud/bigtable/encryption.py
Comment (kolea2) [inline google/cloud/bigtable/encryption.py:20]: for clarity, I would rename this file to `encryption_info.py`

File: google/cloud/bigtable/error.py
Comment (kolea2) [inline google/cloud/bigtable/error.py:23]: is it possible to use the pb2 Status as is? Would be nice to have less handwritten code to look after :)

File: google/cloud/bigtable/cluster.py
Comment (kolea2) [inline google/cloud/bigtable/cluster.py:381]: same question here, what does this look like when kms key is not set? Is there a default view?

File: google/cloud/bigtable/cluster.py
Comment (chrisrossi) [reply google/cloud/bigtable/cluster.py:163]: Oops. If it's being called from the constructor, it will be None, but we can't assume that can we? I'll add an `else` here.

File: google/cloud/bigtable/error.py
Comment (chrisrossi) [reply google/cloud/bigtable/error.py:23]: I was doing it in the first draft, then decided for consistency I should probably wrap it. I think I saw in the Java reference implementation it was wrapped and figured I'd better follow suit. I think my personal preference would be to return the bare pb2, if that's not considered a violation of the contract for user exposed APIs.

Comment (chrisrossi) [reply google/cloud/bigtable/error.py:23]: It also just occurred to me, that the wrapper class might be preferable for documentation purposes, as it causes the interface to get documented in the Sphinx docs. Otherwise, the user has to track down the protocol buffer definitions. 

I'll defer to whatever call you want to make here.

File: google/cloud/bigtable/cluster.py
Comment (chrisrossi) [reply google/cloud/bigtable/cluster.py:381]: When I look at the PBs returned from the service, I'm finding that it's omitting `encryption_config` altogether if using Google default encryption, so this creates a PB the same as we get them from the service.

File: google/cloud/bigtable/encryption.py
Comment (chrisrossi) [reply google/cloud/bigtable/encryption.py]: 090080e

File: google/cloud/bigtable/cluster.py
Comment (chrisrossi) [reply google/cloud/bigtable/cluster.py:163]: f1aae9d

File: google/cloud/bigtable/error.py
Comment (kolea2) [reply google/cloud/bigtable/error.py:23]: Sorry for the delay here - following up on this. I'll update here when I can :)

File: google/cloud/bigtable/cluster.py
Comment (kolea2) [reply google/cloud/bigtable/cluster.py:163]: oh sorry missed that, makes sense thanks!

File: google/cloud/bigtable/error.py
Comment (chrisrossi) [reply google/cloud/bigtable/error.py:23]: Oh no, I'm sorry for the delay in getting back to you. I had fumbled something in my email filters and didn't see you had responded to the initial draft, so I thought I was still waiting.

File: google/cloud/bigtable/cluster.py
Comment (chrisrossi) [reply google/cloud/bigtable/cluster.py:163]: Actually, you correctly pointed out a logical error, which I corrected. Thank you!

File: google/cloud/bigtable/error.py
Comment (kolea2) [reply google/cloud/bigtable/error.py:23]: I think we're going to keep this - resolving

File: google/cloud/bigtable/error.py
Comment (kolea2) [inline google/cloud/bigtable/error.py:38]: nit: typo "google"

File: tests/unit/test_table.py
Comment (kolea2) [inline tests/unit/test_table.py:566]: 😄 , we may want to make these a little more realistic - something like "status ok"

**Review by kolea2 — COMMENTED** [review]
Looking good with two small comments. Will also ask @crwilcox to take a quick look

File: google/cloud/bigtable/error.py
Comment (chrisrossi) [reply google/cloud/bigtable/error.py]: e7a45ee

File: tests/unit/test_table.py
Comment (chrisrossi) [reply tests/unit/test_table.py]: e7a45ee

**Review by kolea2 — APPROVED** [review]
lgtm, will merge after @crwilcox takes a look. Thank you!

**Review by crwilcox — APPROVED** [review]

**Pull request merged**
